### PR TITLE
:tada: Add new dashboard libraries screen

### PR DESCRIPTION
**Pull request by hirunatan**
https://tree.taiga.io/project/penpot/task/3897

Design at https://pre.penpot.app/#/workspace/cfd4ca10-ef85-11eb-a311-6a3557fe78ff/30047eb0-01b6-11ed-a6e9-41df3d3e473f?page-id=839f3c40-01b6-11ed-b178-fda96383448f

### CODE REVIEW

File: exporter/shadow-cljs.edn [hunk 1]
```diff
@@ -12,7 +12,7 @@
               :after-load app.core/start}
 
    :compiler-options
-   {:output-feature-set :es8
+   {:output-feature-set :es2020
```
Comment (superalex) [inline exporter/shadow-cljs.edn]: I think it would be cleaner correct this out of this MR

Comment (hirunatan) [reply exporter/shadow-cljs.edn]: True. I'm going to do this in a separate commit, direct to develop.

File: backend/src/app/rpc/queries/files.clj [hunk 2]
```diff
@@ -420,7 +421,28 @@
 
 (sv/defmethod ::team-shared-files
   [{:keys [pool] :as cfg} {:keys [team-id] :as params}]
-  (db/exec! pool [sql:team-shared-files team-id]))
+  (let [assets-sample
+        (fn [assets limit]
+          (let [sorted-assets (->> assets
+                                   (vals)
+                                   (vec)
+                                   (sort-by #(str/lower (:name %))))]
```
Comment (niwinz) [inline backend/src/app/rpc/queries/files.clj:427]: Can be simplified with:

```clojure
(->> (vals assets)
     (sort-by (comp str/lower :name)))
```
You don't need call `vec` here.

File: backend/src/app/rpc/queries/files.clj [hunk 3]
```diff
@@ -420,7 +421,28 @@
...
+
+          {:count (count sorted-assets)
+           :sample (take limit sorted-assets)}))
```
Comment (niwinz) [inline backend/src/app/rpc/queries/files.clj]: On the end, if you need a vector, `(into [] (take limit) sorted-assets)`

File: frontend/src/app/main/ui/dashboard/grid.cljs [hunk 4]
```diff
@@ -56,6 +58,88 @@
...
+  [:div.grid-item-th.library
+   (if (nil? file)
+     i/loader-pencil
+     (let [summary (:library-summary file)
+           components (:components summary)
+           colors (:colors summary)
+           typographies (:typographies summary)]
+       [:*
+
+        (when (> (:count components) 0)
```
Comment (niwinz) [inline frontend/src/app/main/ui/dashboard/grid.cljs]: `pos?` predicate is preferred

File: frontend/src/app/main/ui/dashboard/grid.cljs [hunk 5]
```diff
@@ -56,6 +58,88 @@
...
+                                  :objects (:objects component)}]
+               [:div.name-block
+                [:span.item-name {:title (:name component)}
+                 (:name component)]]])
+            (when (> (:count components) (count (:sample components)))
+              [:div.asset-list-item
+               [:div.name-block
+                [:span.item-name "(...)"]]])]])
+
+        (when (> (:count colors) 0)
```
Comment (niwinz) [inline frontend/src/app/main/ui/dashboard/grid.cljs]: The same here, `pos?` over `>`

File: frontend/src/app/main/ui/dashboard/grid.cljs [hunk 6]
```diff
@@ -56,6 +58,88 @@
...
+           [:div.asset-title
+            [:span (tr "workspace.assets.colors")]
+            [:span.num-assets (str "\u00A0(") (:count colors) ")"]] ;; Unicode 00A0 is non-breaking space
+           [:div.asset-list
+            (for [color (:sample colors)]
+              (let [default-name (cond
+                                   (:gradient color) (bc/gradient-type->string (get-in color [:gradient :type]))
+                                   (:color color) (:color color)
+                                   :else (:value color))]
+                [:div.asset-list-item
```
Comment (niwinz) [inline frontend/src/app/main/ui/dashboard/grid.cljs]: missing `:key` prop on the `div`

File: frontend/src/app/main/ui/dashboard/grid.cljs [hunk 7]
```diff
@@ -56,6 +58,88 @@
...
+                [:span.item-name "(...)"]]])]])
+
+        (when (> (:count typographies) 0)
+          [:div.asset-section
+           [:div.asset-title
+            [:span (tr "workspace.assets.typography")]
+            [:span.num-assets (str "\u00A0(") (:count typographies) ")"]] ;; Unicode 00A0 is non-breaking space
+           [:div.asset-list
+            (for [typography (:sample typographies)]
+              [:div.asset-list-item
```
Comment (niwinz) [inline frontend/src/app/main/ui/dashboard/grid.cljs]: Missing `:key`

File: frontend/src/app/main/ui/dashboard/grid.cljs [hunk 8]
```diff
@@ -56,6 +58,88 @@
...
+       [:*
+
+        (when (> (:count components) 0)
+          [:div.asset-section
+           [:div.asset-title
+            [:span (tr "workspace.assets.components")]
+            [:span.num-assets (str "\u00A0(") (:count components) ")"]] ;; Unicode 00A0 is non-breaking space
+           [:div.asset-list
+            (for [component (:sample components)]
+              [:div.asset-list-item
```
Comment (niwinz) [inline frontend/src/app/main/ui/dashboard/grid.cljs]: Missing `:key`

File: backend/src/app/rpc/queries/files.clj (hunk 2 above)
Comment (hirunatan) [reply backend/src/app/rpc/queries/files.clj:427]: ok

File: backend/src/app/rpc/queries/files.clj (hunk 3 above)
Comment (hirunatan) [reply backend/src/app/rpc/queries/files.clj]: ok

File: frontend/src/app/main/ui/dashboard/grid.cljs (hunk 4 above)
Comment (hirunatan) [reply frontend/src/app/main/ui/dashboard/grid.cljs]: ok

File: frontend/src/app/main/ui/dashboard/grid.cljs (hunk 5 above)
Comment (hirunatan) [reply frontend/src/app/main/ui/dashboard/grid.cljs]: ok

File: frontend/src/app/main/ui/dashboard/grid.cljs (hunk 6 above)
Comment (hirunatan) [reply frontend/src/app/main/ui/dashboard/grid.cljs]: ok

File: frontend/src/app/main/ui/dashboard/grid.cljs (hunk 8 above)
Comment (hirunatan) [reply frontend/src/app/main/ui/dashboard/grid.cljs]: ok

File: frontend/src/app/main/ui/dashboard/grid.cljs (hunk 7 above)
Comment (hirunatan) [reply frontend/src/app/main/ui/dashboard/grid.cljs]: ok

**Pull request merged**